lib/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.cpp

```diff
@@ -323,11 +324,23 @@ struct ExternElementwiseOpConversion
     if (funcName.empty())
       llvm::errs() << "ExternElementwiseOpConversion";
 
-    Type funcType = getFunctionType(elemTy, operands[0]);
-    LLVM::LLVMFuncOp funcOp = appendOrGetExternFuncOp(
-        rewriter, op, funcName, funcType, op.getLibname(), op.getLibpath());
-    return {
-        rewriter.create<LLVM::CallOp>(loc, funcOp, operands[0]).getResult()};
+    if (op->hasAttr(UseLLVMIntrinsicAttr::getMnemonic())) {
```

Comment:
Hmm, this feels a bit hacky. Actually I think we don't need to go all the way down to llvm intrinsics right here if device lib doesn't have a symbol for the one we want and we need to resort to llvm intrinsics. You can think of it as a multi-step lowering in mlir commonly. Here we can "fake" a device lib with `__ocml_trition_*` function call and then later "lower" that function call into a corresponding llvm intrinsics. the later step can be put in [BuiltinFuncToLLVM.cpp](https://github.com/openai/triton/blob/main/third_party/amd/lib/TritonAMDGPUToLLVM/BuiltinFuncToLLVM.cpp) like https://github.com/openai/triton/pull/3767. This avoids hacks here.

Reply:
Hi @antiagainst, 

> Here we can "fake" a device lib with __ocml_trition_* function call and then later "lower" that function call into a corresponding llvm intrinsics.

Done, can you have a look please?

I also reverted some files.